packages/editor/src/store/reducer.js

```diff
@@ -361,63 +428,114 @@ export const editor = flow( [
 		byClientId( state = {}, action ) {
 			switch ( action.type ) {
 				case 'SETUP_EDITOR_STATE':
-					return getFlattenedBlocks( action.blocks );
+					return getFlattenedBlocksWithoutAttributes( action.blocks );
 
 				case 'RECEIVE_BLOCKS':
 					return {
 						...state,
-						...getFlattenedBlocks( action.blocks ),
+						...getFlattenedBlocksWithoutAttributes( action.blocks ),
 					};
 
-				case 'UPDATE_BLOCK_ATTRIBUTES':
+				case 'UPDATE_BLOCK':
 					// Ignore updates if block isn't known
 					if ( ! state[ action.clientId ] ) {
 						return state;
 					}
 
-					// Consider as updates only changed values
-					const nextAttributes = reduce( action.attributes, ( result, value, key ) => {
-						if ( value !== result[ key ] ) {
-							result = getMutateSafeObject( state[ action.clientId ].attributes, result );
-							result[ key ] = value;
-						}
-
-						return result;
-					}, state[ action.clientId ].attributes );
-
-					// Skip update if nothing has been changed. The reference will
-					// match the original block if `reduce` had no changed values.
-					if ( nextAttributes === state[ action.clientId ].attributes ) {
+					// Do nothing if only attributes change.
```

Comment:
What does / should this do if the update is for `innerBlocks`?

Reply:
Is it valid semantics to modify inner blocks with an `UPDATE_BLOCK` instead of `MOVE_*` and other actions? I assumed not, since that possibility isn't covered in the existing code.

Reply:
While I think it's something we should respect, we already don't have consideration for a changed `innerBlocks` from the `order` subtree, so I think it'd be fine to defer.